update-docs.ps1

```diff
@@ -22,18 +22,31 @@ param (
 . ./shared.ps1
 
 function Update-Canonical-Data {
-    Write-Output "Updating canonical data"
-    Run-Command "./update-canonical-data.ps1" 
+    [CmdletBinding(SupportsShouldProcess)]
+    param ()
+
+    if ($PSCmdlet.ShouldProcess($true)) {
+        Write-Output "Updating canonical data"
+        Invoke-CommandExecution "./update-canonical-data.ps1" 
+    }
 }
 
-function Update-Docs {
-    Write-Output "Updating docs"
-    $args = if ($Exercise) { @("-o", $Exercise) } else { @() }
-    Run-Command "./bin/fetch-configlet"
-    Run-Command "./bin/configlet generate . -p problem-specifications $args"
+function Update-DocFile {
+    [CmdletBinding(SupportsShouldProcess)]
+    param (
+        [Parameter(Position = 0, Mandatory = $false)]
+        [string]$Exercise
+    )
+
+    if ($PSCmdlet.ShouldProcess($true)) {
```

Comment:
What if: Performing the operation "fetch configlet and pull specifications" on target "All Exercises".

`Update-DocsFolder"example" -WhatIf`
`What if: Performing the operation "fetch configlet and pull specifications" on target "example"`

Reply:
That's really nice. I think you forgot to add `[CmdletBinding(SupportsShouldProcess)]` to the top-level `param`, as otherwise it won't flow to this sub-task.